test/e2e/setup.sh

```diff
@@ -6,7 +6,7 @@ set -o pipefail
 
 # Before updating the default kind image to use, verify that the version is supported
 # by the current kind release.
-KIND_IMAGE="${KIND_IMAGE:-kindest/node:v1.28.0}"
```

Comment:
If updating the version here, we could update the version in the [Makefile](https://github.com/Azure/fleet/blob/main/Makefile) and [ReadMe](https://github.com/Azure/fleet/blob/main/README.md)

Reply:
I will leave this to another PR